src/zkscript/util/utility_scripts.py

```diff
@@ -130,3 +137,59 @@ def nums_to_script(nums: list[int]) -> Script:
             out.append_pushdata(encode_num(n))
 
     return out
+
+
+def mod(
+    is_tuck: bool = True, is_from_alt: bool = True, is_positive: bool = True, is_constant_reused: bool = True
+) -> Script:
+    """Perform modulo operation in Bitcoin Script.
+
+    This function generates a Bitcoin Script that performs a modulo operation. The behavior of the
+    operation can be customized using the provided parameters.
+
+    Args:
+        is_tuck (bool, optional): If True, uses OP_TUCK; if False, uses OP_OVER. Defaults to True.
+        is_from_alt (bool, optional): If True, the value is loaded from the altstack. Defaults to True.
+        is_positive (bool, optional): If True, adds operations to ensure the modulo value is positive.
+            Defaults to True.
+        is_constant_reused (bool, optional): If True, modifies the script to leave the modulo constant in the stack.
+            Defaults to True.
+
+    Returns:
+        Script: A Bitcoin Script that performs the modulo operation based on the specified parameters.
```

Comment:
I think the documentation needs some examples to explain the behaviour of the function and the expected stack given the parameters with which the function is called

Reply:
should be clearer now, let me know if it needs more work!

Reply:
Yes, much better. I think it might be worth pointing out in the docs that the meaning of `is_tuck` is: if `is_tuck = True`, then the modulo is the top element of the stack, while if `is_tuck = False`, then the modulo is the second to top element of the stack.